### PR TITLE
Disable Kubernetes Dashboard by default

### LINKED ISSUE #2547: google_container_cluster - kubernetes_dashboard should be disabled by default
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment. If the issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If the issue is assigned to a user, that user is claiming responsibility for the issue. If the issue is assigned to "hashibot", a community member has claimed the issue already.

### Description

GKE dashboard [is deprecated](https://cloud.google.com/kubernetes-engine/docs/concepts/dashboards) but it [is enabled by default](https://www.terraform.io/docs/providers/google/r/container_cluster.html#kubernetes_dashboard). It should be disabled by default.

### New or Affected Resource(s)

* google_container_cluster

### References

https://cloud.google.com/kubernetes-engine/docs/concepts/dashboards

### PR DESCRIPTION
**Pull request by vbhavsar**
Kubernetes Dashboard has been deprecated and should not be enabled by default.

Fixes #2547.

### CODE REVIEW

**Comment by vbhavsar** [comment]
cc @ndmckinley

File: google/resource_container_cluster.go [hunk 1]
```diff
@@ -1597,6 +1597,12 @@ func flattenClusterAddonsConfig(c *containerBeta.AddonsConfig) []map[string]inte
 				"disabled": c.KubernetesDashboard.Disabled,
 			},
 		}
+	} else {
```
Comment (ndmckinley) [inline google/resource_container_cluster.go]: Unfortunately this is backwards - flatteners go from api objects to terraform objects, but expanders go from terraform objects to api objects.  You'll need to make a similar change in the expander.  Any reason you didn't choose to set the default in the schema object instead?

**Comment by ndmckinley** [comment]
That would be interesting!  I'll run that through the generation system and run the tests on it.

**Comment by ndmckinley** [comment]
Ah, right - that doesn't compile, k8s doesn't have the `Enabled` field on their dashboard object.

**Comment by vbhavsar** [comment]
@ndmckinley Sorry, I didn't intend for this change to be reviewed yet. I did notice that k8s didn't have `Enabled`. I can make a pull request to k8s if you think it is a worthy change and likely to be accepted. 

An alternate is to set `Disabled` to false by default.

**Comment by ndmckinley** [comment]
Hi,

That might work!  I think that setting it to false by default has a chance of working depending on how the underlying api calls wind up working.  Are you comfortable implementing and testing that?

I'm confident k8s won't accept that PR - their client code is autogenerated.

**Comment by vbhavsar** [comment]
> An alternate is to set `Disabled` to false by default.

s/false/true/ but I think you know what I meant. :) 

> Are you comfortable implementing and testing that?

I will give it a shot.

**Comment by danawillow** [comment]
Hi @vbhavsar, sorry for the delay! I'll be taking over this review while Nathan is out.

Have you tested this change? Just looking at the code, I think it'll only work successfully if the user specifies an empty `kubernetes_dashboard` block.

**Comment by vbhavsar** [comment]
Hi @danawillow. I should have posted a comment here when I tried and failed to implement your suggestion.

I'm happy to know that the documentation is useful at least. :) 

Thanks for merging!

**Pull request merged**